src/maker.rs

```diff
@@ -468,4 +551,35 @@ mod tests {
         assert!(reaction.is_some());
         assert_eq!(maker.mid_market_rate, Some(new_mid_market_rate))
     }
+
+    #[test]
+    fn free_funds_and_remove_ongoing_taker_when_processing_finished_swap() {
+        let mut maker = Maker {
+            btc_reserved_funds: bitcoin::Amount::from_btc(1.1).unwrap(),
+            dai_reserved_funds: dai::Amount::from_dai_trunc(1.1).unwrap(),
+            btc_fee: bitcoin::Amount::from_btc(0.1).unwrap(),
+            dai_fee: dai::Amount::from_dai_trunc(0.1).unwrap(),
+            ..Default::default()
+        };
+
+        let taker = Taker { 0: 0 };
+
+        maker.ongoing_takers.insert(taker).unwrap();
```

Comment:
:upside_down_face: Perhaps it would be better to `process_taken_order` as part of the test instead of building the state manually. Or not. Just an idea.

Reply:
That's what I had in mind with crating more elaborate roundtrip tests. I can also start doing it here. I tried to only test one function per test so far so it is clear what is tested.